Switch to using MkDocs

relh opened the pull request:
I have successfully implemented a comprehensive table of contents and documentation index system for the Metta AI project:

Changes Made:
Main README.md - Added a comprehensive table of contents that includes:
Internal section links (Introduction, Installation, Usage, etc.)
Documentation hub links (API, Map Generation, Workflows)
Component documentation links (Agent, Common, MettagGrid, etc.)
Project resource links (Roadmap, Configuration, DevOps)
INDEX.md - Created an alphabetical index of key concepts with links to:
Configuration documentation
Source code locations
API references
Specific sections within documentation files

docs/README.md - Created a central documentation hub that:
Organizes all documentation into logical categories
Provides clear descriptions of each document's purpose
Includes a documentation structure diagram
Offers getting started guidance for different user types
docs/quick-guide.md - Created a concise quick-start guide that provides:
5-minute installation instructions
First training run examples
Visualization options
Key concepts overview
Common troubleshooting tips

Enhanced existing documentation with tables of contents:
docs/api.md - Added TOC for API sections
docs/wandb/metrics/README.md - Added TOC for metrics documentation
roadmap.md - Added TOC for research areas

Documentation Organization:
The documentation is now organized into clear categories:
Core Documentation: Quick guide, API reference, map generation
Automation & Workflows: GitHub Actions and CI/CD documentation
Metrics & Monitoring: WandB metrics documentation
Component Documentation: Links to module-specific READMEs
Project Resources: Roadmap, configuration, and infrastructure docs
This structure significantly improves documentation navigability by providing multiple entry points (table of contents, alphabetical index, documentation hub) and clear hierarchical organization.

[Asana Task](https://app.asana.com/1/1209016784099267/project/1210348820405981/task/1210767321451330)

nishu-builder reviewed (changes requested):
This looks great, especially the in-existing-doc tables of contents

I dont love the proliferation of docs (docs/index.md) and how it would force us to update multiple places whenever we update docs

Replace the master index with auto-generation - Instead of a manually maintained docs/index.md, consider:

Could we manage things like the index with a script that auto-generates the index from existing docs? could use tools like MkDocs or Docusaurus that handles navigation automatically

relh commented:
# Replace Manual Documentation Index with MkDocs Auto-Generation

This PR addresses the feedback from [PR #1412](https://github.com/Metta-AI/metta/pull/1412) about manual documentation maintenance by implementing MkDocs as an automatic documentation generator.

## Problem
The previous approach required:
- Manual maintenance of `docs/index.md` alphabetical index
- Manual table of contents in each documentation file
- Updates in multiple places when documentation changed

## Solution
Implemented MkDocs with automatic generation features:
- **Full-text search** replaces manual alphabetical index
- **Automatic table of contents** generated from markdown headers
- **Dynamic navigation** that updates when files are added/removed

## Changes Made

### 1. Added MkDocs Configuration
- `mkdocs.yml` - Site configuration with Material theme
- Automatic TOC generation for all pages
- Full-text search functionality
- Navigation structure (with auto-discovery option)

### 2. Updated Dependencies
Added to `pyproject.toml`:
```toml
"mkdocs>=1.5.0",
"mkdocs-material>=9.5.0", 
"mkdocs-gen-files>=0.5.0",
"mkdocstrings[python]>=0.24.0",
```

### 3. CLI Integration
New commands in `metta` CLI:
```bash
metta docs serve    # Live preview with auto-reload
metta docs build    # Build static site  
metta docs deploy   # Deploy to GitHub Pages
```

### 4. Auto-Discovery Script
- `scripts/gen_docs_nav.py` - Automatically discovers documentation files
- Builds navigation from directory structure
- No manual updates needed when files change

### 5. GitHub Actions Workflow
- `.github/workflows/docs.yml` - Automatic deployment to GitHub Pages
- Builds documentation on every push to main
- Preview builds for PRs

### 6. Removed Manual Index
- Deleted `docs/index.md` - replaced by MkDocs search functionality
- No more manual maintenance required

## Benefits

1. **Zero Manual Maintenance**
   - No need to update index when files are added/removed
   - Navigation updates automatically
   - Search index rebuilds on every build

2. **Better Features**
   - Full-text search across all documentation
   - Automatic table of contents with permalinks
   - Live reload during development
   - Mobile-responsive design

3. **Simple Migration**
   - All existing markdown files work as-is
   - No need to rewrite documentation
   - Can remove manual TOCs from existing files

## Usage

```bash
# Install dependencies
uv sync

# Preview documentation locally
metta docs serve

# Build static site
metta docs build

# Deploy to GitHub Pages
metta docs deploy
```

Visit http://localhost:8000 to see the documentation site.

## Migration Notes

- Existing documentation files require no changes
- Manual TOCs in files can be removed (MkDocs generates them)
- The alphabetical index is replaced by search functionality
- All existing links and references continue to work

This approach eliminates the "proliferation of docs" concern by providing automatic generation instead of manual maintenance, ensuring documentation stays in sync with the codebase without additional effort.
